hypershift-operator/metrics.go

```diff
@@ -178,7 +202,36 @@ func (m *hypershiftMetrics) observeHostedClusters(hostedClusters *hyperv1.Hosted
 	hcCount := newLabelCounter()
 	hcByConditions := newLabelCounter()
 
+	identityProvidersCounter := make(map[string]float64)
 	for _, hc := range hostedClusters.Items {
+		// Collect proxy metric.
+		var proxyHTTP, proxyHTTPS, proxyTrustedCA string
+		if hc.Spec.Configuration != nil && hc.Spec.Configuration.Proxy != nil {
+			if hc.Spec.Configuration.Proxy.HTTPProxy != "" {
+				proxyHTTP = "1"
+			}
+			if hc.Spec.Configuration.Proxy.HTTPSProxy != "" {
+				proxyHTTPS = "1"
+			}
+			if hc.Spec.Configuration.Proxy.TrustedCA.Name != "" {
+				proxyTrustedCA = "1"
+			}
+			m.clusterProxy.WithLabelValues(hc.Namespace, hc.Name, proxyHTTP, proxyHTTPS, proxyTrustedCA).Set(1)
+		}
+
+		// Group identityProviders by type.
+		if hc.Spec.Configuration != nil && hc.Spec.Configuration.OAuth != nil {
```

Comment:
For identity providers, you should always emit a value for each possible label.
See 
https://github.com/openshift/osd-metrics-exporter/blob/be64e94da0caa0ba7941324084b4b14ba97c70c6/pkg/metrics/aggregator.go#L144-L150

Reply:
This is counting each idp for each hc and grouping them.

So then we set the metric as in your link here https://github.com/openshift/hypershift/pull/2015/files#diff-f9d495981021f385c74344cb3c9596ab156206f89cf88a847a071c15b2b2d9eaR269

Reply:
but if you don't have any idps you should still report 0 for every possible value.